fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSessionTest.java

```diff
@@ -195,4 +223,243 @@ public void getTriplesOfBinaryDescription() throws Exception {
+            //expected failure
+        }
+    }
+
+    /**
+     * This test covers the expected behavior when two OCFL Object Sessions are modified and one of the commits to
+     * the mutable head fails.
+     *
+     * @throws PersistentStorageException
+     */
+    @Test
+    public void rollbackOnSessionWithCommitsToMutableHeadShouldFail() throws PersistentStorageException {
+        when(mapping.getOcflObjectId()).thenReturn(OCFL_OBJECT_ID);
+        when(mapping.getParentFedoraResourceId()).thenReturn(parentId);
+        when(index.getMapping(resourceId)).thenReturn(mapping);
+
+        final String ocfObjectId2 = OCFL_OBJECT_ID + "2";
+        when(mapping2.getOcflObjectId()).thenReturn(ocfObjectId2);
+        when(mapping2.getParentFedoraResourceId()).thenReturn(resourceId2);
+        when(index.getMapping(resourceId2)).thenReturn(mapping2);
+
+        final Node resourceUri = createURI(resourceId);
+
+        when(rdfSourceOperation.getTriples()).thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getServerManagedProperties())
+                .thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getResourceId()).thenReturn(resourceId);
+        when(rdfSourceOperation.getType()).thenReturn(CREATE);
+
+        final Node resourceUri2 = createURI(resourceId2);
+
+        when(rdfSourceOperation2.getTriples()).thenReturn(new DefaultRdfStream(resourceUri2, Stream.empty()));
+        when(rdfSourceOperation2.getServerManagedProperties())
+                .thenReturn(new DefaultRdfStream(resourceUri2, Stream.empty()));
+        when(rdfSourceOperation2.getResourceId()).thenReturn(resourceId2);
+        when(rdfSourceOperation2.getType()).thenReturn(CREATE);
+
+        //mock success on commit for the first object session
+        when(mockSessionFactory.create(eq(OCFL_OBJECT_ID), anyString())).thenReturn(objectSession1);
+        when(objectSession1.commit(eq(CommitOption.UNVERSIONED))).thenReturn(Instant.now().toString());
+        //mock failure on commit for the second object session
+        when(mockSessionFactory.create(eq(ocfObjectId2), anyString())).thenReturn(objectSession2);
+        when(objectSession2.commit(eq(CommitOption.UNVERSIONED))).thenThrow(PersistentStorageException.class);
+
+        final PersistentStorageSession session1 = createSession(index, mockSessionFactory);
+        //persist the two operations
+        session1.persist(rdfSourceOperation);
+
+        //perform the create rdf operation
+        session1.persist(rdfSourceOperation2);
+
+        //get triples should now fail because the session is effectively closed.
+        try {
+            session1.commit(CommitOption.UNVERSIONED);
+            fail("session1.commit(...) invocation should fail.");
+        } catch (PersistentStorageException ex) {
+            //attempted rollback should also fail:
+            try {
```

Comment:
this doesn't need to be inside the catch block

Reply:
Are you sure?

Reply:
Pretty sure, if no exception is thrown then it will fail first, and if any exception other than the expected one is thrown the the test will also fail. But its mainly cosmetic, so it doesn't matter too much